### PR TITLE
nixos/miniflux: use systemd notify and watchdog

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Description of changes
Set systemd service type to notify, as miniflux supports type=notify [since March 2021 (2.0.29)](https://github.com/miniflux/v2/blob/a631bd527d73b5b8d8e85343a1d5fe438e89dad7/ChangeLog#L842).
Enable the systemd watchdog, which miniflux supports [since June 2021 (2.0.31)](https://github.com/miniflux/v2/blob/a631bd527d73b5b8d8e85343a1d5fe438e89dad7/ChangeLog#L806).

I also removed the usage of `with lib;` in the miniflux module.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/web-apps/miniflux.nix [hunk 1]
```diff
@@ -96,12 +96,18 @@ in
...
-        ExecStart = "${cfg.package}/bin/miniflux";
+        Type = "notify";
+        ExecStart = lib.getExe cfg.package;
         User = "miniflux";
         DynamicUser = true;
         RuntimeDirectory = "miniflux";
         RuntimeDirectoryMode = "0750";
         EnvironmentFile = cfg.adminCredentialsFile;
+        WatchdogSec = 60;
+        WatchdogSignal = "SIGKILL";
```
Comment (getchoo) [inline nixos/modules/services/web-apps/miniflux.nix:107]: everything else looks fine here, but i do have to ask: why not stick with systemd's default of `SIGABRT` here? does miniflux not handle this well?` SIGKILL` shouldn't be really be used in most situations

**Review by getchoo — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/miniflux.nix (hunk 1 above)
Comment (NyCodeGHG) [reply nixos/modules/services/web-apps/miniflux.nix:107]: I copied this from upstreams systemd unit: https://github.com/miniflux/v2/blob/a631bd527d73b5b8d8e85343a1d5fe438e89dad7/packaging/systemd/miniflux.service#L26
I don't know how it behaves when using SIGABRT, I'll have a look.

Comment (NyCodeGHG) [reply nixos/modules/services/web-apps/miniflux.nix:107]: The go runtime prints the stack trace of all goroutines and dumps registers.
Looks like it works fine when using SIGABRT.

Comment (getchoo) [reply nixos/modules/services/web-apps/miniflux.nix:107]: > I copied this from upstreams systemd unit:

that's good enough for me. thanks!

**Review by getchoo — APPROVED** [review]

**Comment by NyCodeGHG** [comment]
I set WATCHDOG=1 in the config to make sure the watchdog is explicitly enabled (even though that setting is the default).

**Comment by NyCodeGHG** [comment]
@ofborg test miniflux

**Review by getchoo — APPROVED** [review]

**Pull request merged**